lib/practice_exercises.rb

```diff
@@ -1,13 +1,34 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  return true if string.empty? || string.length == 1
+  string.downcase!
+  split_string = string.split("")
+  char_string = split_string.select{ |char| /[a-z]/.match?(char) }
+
+  half = char_string.length / 2
+  half.times do |i|
+    return false if char_string[i] != char_string[(char_string.length - 1) - i]
+  end
+
+  return true
+  # raise NotImplementedError, "Not implemented yet"
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: 0(1)
 def longest_prefix(strings)
```

Comment:
I would say that this is O(n * m) time complexity where m is the number of letters in the shortest word.

Otherwise 👍